### ISSUE TITLE
OAuth2 flow improvement in re-authorization

### ISSUE THREAD

**Issue by jesmrec** [owncloud/ios-app#293]
Some improvements suggested here: https://github.com/owncloud/ios-app/pull/291#issuecomment-468204896

- It would be cool that is not needed to reload the account to see the non-authorization alert when the token is dismissed in dashboard. Currently, it is only displayed the `Offline. Contents from cache` notification status message.

- If a server changes its authentication method from OAuth2 to basic, edition of credentials is not posible since username is not editable. Remove the account and add it again is the only option to skip. This is an open discussion, so this is not a bad behaviour. Ideas welcome.

**Comment by T0mWz** [comment]
When the oAuth session has expired, the client does not restart an oAuth flow, but tries Basic Authentication.
However, in our case we use Shibboleth, where Basic Auth does not work, so the connection must be completely removed and added again.
See also the log message; `No public access to this resource., No 'Authorization: Basic' header found. Either the client didn't send one, or the server is misconfigured, Bearer token was incorrect, No 'Authorization: Basic' header found. Either the client didn't send one, or the server is misconfigured`

Please, could you take a look at this? Have added some logging below, not sure if this is useful;

```
...
2019-07-02 09:04:30.058000+0200 ownCloud File Provider[1318.373129] [dbug] | [IPNC] Received notification 'org.owncloud.log-component-enabled:writer.stderr' [OCIPNotificationCenter.m:168|FULL]
2019-07-02 09:04:30.058000+0200 ownCloud File Provider[1318.373129] [dbug] | [IPNC] Delivering notification 'org.owncloud.log-component-enabled:writer.stderr' to <OCLogWriter: 0x10e202bf0> [OCIPNotificationCenter.m:198|FULL]
2019-07-02 09:04:30.058000+0200 ownCloud File Provider[1318.373129] [dbug] | [IPNC] Received notification 'org.owncloud.log-settings-changed' [OCIPNotificationCenter.m:168|FULL]
2019-07-02 09:04:30.058000+0200 ownCloud File Provider[1318.373129] [dbug] | [IPNC] Delivering notification 'org.owncloud.log-settings-changed' to <OCLogger: 0x10bf165d0> [OCIPNotificationCenter.m:198|FULL]
2019-07-02 09:04:30.063000+0200 ownCloud[1496:404621] [dbug] | [HTTP, Background, …] Task [taskIdentifier=<1>, url=https://surfdrive.surf.nl/files/remote.php/dav/files/<account-here>/1499630604.png] didFinishCollectingMetrics: { total: [2019-07-02 07:04:29 +0000 - 2019-07-02 07:04:30 +0000, 0.23 sec], startedAfter: 0.00, redirects: 0, transactions: [1: fetchStart: 0.06, DNS: 0.02..0.02 (0.00), connect: 0.02..0.06 (0.05), TLS: 0.03..0.06 (0.04), request: 0.06..0.07 (0.00), cloud: 0.07..0.12 (0.06)], [2: fetchStart: 0.13, DNS: 0.14..0.14 (0.00), connect: 0.14..0.15 (0.01), TLS: 0.15..0.15 (0.01), request: 0.15..0.15 (0.00), cloud: 0.15..0.23 (0.07), response: 0.23..0.23 (0.00)] } [… PipelineID:background, Instance:0x2836c0000, URLSessionID:background;com.owncloud.ios-app, HTTP, Metrics, GET, RequestID:01180ADA-7582-45D1-AAE2-CB9BF2C88DE0, URLSessionTaskID:1] [OCHTTPPipeline.m:1669|FULL]
2019-07-02 09:04:30.064000+0200 ownCloud File Provider[1318:404702] [dbug] | [SQL] Changed memory limit from 1048576 to 1048576 bytes [OCSQLiteDB.m:771|FULL]
2019-07-02 09:04:30.074000+0200 ownCloud[1496:404551] [dbug] | [HTTP, Background, …] https://surfdrive.surf.nl/files/remote.php/dav/files/<account-here>/1499630604.png [taskIdentifier=<1>]: downloadTask:didFinishDownloadingToURL: file:///private/var/mobile/Containers/Shared/AppGroup/F3B31130-8CF1-4D04-829A-411F3AEDA699/Library/Caches/com.apple.nsurlsessiond/Downloads/com.owncloud.ios-app/CFNetworkDownload_z73sf7.tmp [… PipelineID:background, Instance:0x2836c0000, URLSessionID:background;com.owncloud.ios-app] [OCHTTPPipeline.m:1915|FULL]
2019-07-02 09:04:30.075000+0200 ownCloud[1496:404551] [dbug] | [HTTP, Background, …] Task [taskIdentifier=<1>, url=https://surfdrive.surf.nl/files/remote.php/dav/files/<account-here>/1499630604.png] didCompleteWithError=(null) [… PipelineID:background, Instance:0x2836c0000, URLSessionID:background;com.owncloud.ios-app] [OCHTTPPipeline.m:1626|FULL]
2019-07-02 09:04:30.075000+0200 ownCloud[1496:404551] [dbug] | [HTTP, Background, …] Known task [taskIdentifier=<1>, url=https://surfdrive.surf.nl/files/remote.php/dav/files/<account-here>/1499630604.png] didCompleteWithError=(null) [… PipelineID:background, Instance:0x2836c0000, URLSessionID:background;com.owncloud.ios-app] [OCHTTPPipeline.m:1630|FULL]
2019-07-02 09:04:30.078000+0200 ownCloud File Provider[1318:404336] [dbug] | [SQL] Changed memory limit from 1048576 to 1048576 bytes [OCSQLiteDB.m:771|FULL]
2019-07-02 09:04:30.079000+0200 ownCloud[1496:404669] [dbug] | Signaling NSFileProviderRootContainerItemIdentifier for changes ended with error (null) [OCVault+Internal.m:373|FULL]
2019-07-02 09:04:30.085000+0200 ownCloud[1496:403417] [dbug] | [HTTP, Background, …] Received response:
# RESPONSE --------------------------------------------------------
Method:     GET
URL:        https://surfdrive.surf.nl/files/remote.php/dav/files/<account-here>/1499630604.png
Request-ID: 01180ADA-7582-45D1-AAE2-CB9BF2C88DE0
Error:      -
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
401 UNAUTHORIZED
Content-Type: application/xml; charset=utf-8
Pragma: no-cache
Content-Security-Policy: default-src 'none';
Set-Cookie: ocff17e49795=4c1154282275e0bd767faf30789aa2b3; path=/files; secure; HttpOnly, oc_sessionPassphrase=qJb3rt8ZKXDsXZOaLjXef0%2Ft4wt4stGulmUyG01i%2B4VRZYRwckFPQHKGUhDuGk6G9wTLsNRkU%2BmqSoPEhxPBPwej0Ggxq0eVM8yZwxjm6Oa6kP3geRRdqE%2B8Gcu7lbrb; path=/files; secure; HttpOnly, app_cookie=web-c-11; path=/; HttpOnly; Secure
Server: Apache
X-Download-Options: noopen
X-XSS-Protection: 1; mode=block
Expires: Thu, 19 Nov 1981 08:52:00 GMT
X-Permitted-Cross-Domain-Policies: none
Cache-Control: no-store, no-cache, must-revalidate
Date: Tue, 02 Jul 2019 07:04:29 GMT
X-Robots-Tag: none
Content-Length: 476
Strict-Transport-Security: max-age=31536000; includeSubDomains;
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
Www-Authenticate: Bearer realm="SURFdrive", Basic realm="SURFdrive", charset="UTF-8"

<?xml version="1.0" encoding="utf-8"?>
<d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns">
  <s:exception>Sabre\DAV\Exception\NotAuthenticated</s:exception>
  <s:message>No public access to this resource., No 'Authorization: Basic' header found. Either the client didn't send one, or the server is misconfigured, Bearer token was incorrect, No 'Authorization: Basic' header found. Either the client didn't send one, or the server is misconfigured</s:message>
</d:error>

----------------------------------------------------------------- [… PipelineID:background, Instance:0x2836c0000, URLSessionID:background;com.owncloud.ios-app, HTTP, Response, GET, RequestID:01180ADA-7582-45D1-AAE2-CB9BF2C88DE0, URLSessionTaskID:1] [OCHTTPPipeline.m:1049|FULL]
2019-07-02 09:04:30.085000+0200 ownCloud[1496:403417] [dbug] | [CORE] Beginning activity 'Handling event' .. [OCCore.m:1429|FULL]
2019-07-02 09:04:30.085000+0200 ownCloud[1496:403417] [dbug] | [HTTP, Background, …] Removing request https://surfdrive.surf.nl/files/remote.php/dav/files/<account-here>/1499630604.png with taskIdentifier <1> [… PipelineID:background, Instance:0x2836c0000, URLSessionID:background;com.owncloud.ios-app] [OCHTTPPipeline.m:1208|FULL]
2019-07-02 09:04:30.085000+0200 ownCloud[1496:404621] [dbug] | [CORE] Beginning activity 'Queuing sync event' .. [OCCore.m:1429|FULL]
2019-07-02 09:04:30.085000+0200 ownCloud[1496:404621] [dbug] | [CORE] Ended activity 'Handling event' .. [OCCore.m:1449|FULL]
2019-07-02 09:04:30.078000+0200 ownCloud File Provider[1318:404336] [dbug] | [FP] Domain set: <NSFileProviderDomain: 0x10be13100, SURFdrive (D5A60ED8-D7CD-4F91-B5D4-2F505A608CA0)> [FileProviderExtension.m:74|FULL]
2019-07-02 09:04:30.078000+0200 ownCloud File Provider[1318:404336] [dbug] | [FP] Domain set: <NSFileProviderDomain: 0x10be13100, SURFdrive (D5A60ED8-D7CD-4F91-B5D4-2F505A608CA0)> [FileProviderExtension.m:74|FULL]
2019-07-02 09:04:30.087000+0200 ownCloud[1496:404622] [dbug] | [CORE, SyncEngine] setNeedsToProcessSyncRecords [OCCore+SyncEngine.m:458|FULL]
2019-07-02 09:04:30.087000+0200 ownCloud[1496:404622] [dbug] | [CORE] Beginning activity 'process sync records if needed' .. [OCCore.m:1429|FULL]
2019-07-02 09:04:30.087000+0200 ownCloud[1496:404622] [dbug] | [CORE] Ended activity 'Queuing sync event' .. [OCCore.m:1449|FULL]
2019-07-02 09:04:30.087000+0200 ownCloud[1496:404498] [dbug] | [CORE, SyncEngine] processSyncRecordsIfNeeded (needed=1) [OCCore+SyncEngine.m:483|FULL]
2019-07-02 09:04:30.087000+0200 ownCloud[1496:404498] [dbug] | [CORE] Beginning activity 'process sync records' .. [OCCore.m:1429|FULL]
...
```

**Comment by michaelstingl** [comment]
Fix got released with the 1.0.2 Saturday morning. @hosy close here?
